src/coreclr/vm/codeman.cpp

```diff
@@ -1541,6 +1541,20 @@ void EEJitManager::SetCpuInfo()
     CPUCompileFlags.EnsureValidInstructionSetSupport();
 
 #if defined(TARGET_X86) || defined(TARGET_AMD64)
+
+    // In case if current CPU supports AVX2 and it's disabled by one of the CLRConfig (e.g. EnableSSE41=0)
```

Comment:
Would it be more logical to allow multiple Vector64/Vector128/Vector256/Vector512 bits before this point (ie delete `CPUCompileFlags.Clear(InstructionSet_VectorT128)` above) and normalize the Vector64/Vector128/Vector256/Vector512 bits here by only keeping the largest one set?

Reply:
`VectorXXX` seem to be fine since they don't exlude each other, right? It's only `VectorT` that is problematic because `InstructionSet_VectorT256` means `InstructionSet_VectorT128` has to be removed (while `Vector256` doesn't remove `Vector128`) if I understood you correctly

Reply:
Yes, I meant `VectorT...`.

Reply:
Did you mean something like I've just pushed?

Reply:
Doing so might require a change in R2R to handle it, but it should be possible.

The only real consideration is that the bits ultimately tracked by the VM (for the purposes of opportunistic/required tracking) and reported to the JIT needs to only have 1 of the 3 set. The ISAs are exclusive to try and avoid issues where a code path may only check one and not the others.